shims/src/peschke/shims/resource/Managed.scala

```diff
@@ -0,0 +1,118 @@
+package peschke.shims.resource
+
+import cats.syntax.all._
+import cats.{Applicative, Monad}
+
+/** This is a poor replacement for cats-effect's `Resource`
+  *
+  * If you have access to cats-effect, use `Resource`. If you don't, this can
+  * serve as a shim.
+  */
+trait Managed[F[_], R] {
+
+  /** Use the resource
+    */
+  def use[Out](f: R => F[Out]): F[Out]
+
+  /** Get the resource
+    *
+    * This is handy when the bulk of the computations have been handled in `map`
+    * and this returns a computed value, rather than the resource itself.
+    *
+    * This can be used to retrieve a resource, however no guarantees can be made
+    * WRT to the validity of the returned value.
+    */
+  def value(implicit A: Applicative[F]): F[R] = use(_.pure[F])
+}
+object Managed {
+
+  def factory[F[_], E]: Factory[F, E] = new Factory[F, E]()
+
+  final class Factory[F[_], E](private val ignored: Boolean = true)
+      extends AnyVal {
+
+    /** Create a `Managed` instance that always returns the same resource.
+      *
+      * Note: because it always returns the same instance, it will never release
+      * this resource.
+      */
+    def pure[R](resource: R)(implicit EM: ExceptionMapper[F, E])
+      : Managed[F, R] = new Pure[F, E, R](resource)
+
+    /** Create a `Managed` instance that always returns a new resource.
+      *
+      * This is a convenience variant of [[Factory.alwaysF()]], which lifts the
+      * returned value into `F`
+      */
+    def always[R]
+      (open:       => R)
+      (implicit C: Closer[F, R], B: Bracket[F], EM: ExceptionMapper[F, E])
+      : Managed[F, R] =
+      new Resource[F, E, R](() => EM.catchNonFatal(open))
+
+    /** Create a `Managed` instance that always returns a new resource.
+      */
+    def alwaysF[R]
+      (open:       => F[R])
+      (implicit C: Closer[F, R], B: Bracket[F], EM: ExceptionMapper[F, E])
+      : Managed[F, R] =
+      new Resource[F, E, R](() => open)
+  }
+
+  implicit def catsInstances[F[_], E](implicit EM: ExceptionMapper[F, E])
+    : Monad[Managed[F, *]] =
+    new Monad[Managed[F, *]] {
+      override def pure[R](x: R): Managed[F, R] = new Pure[F, E, R](x)
+
+      override def flatMap[R0, R1](fa: Managed[F, R0])(f: R0 => Managed[F, R1])
+        : Managed[F, R1] =
+        new FlatMapped[F, E, R0, R1](fa, f)
+
+      override def map[R0, R1](fa: Managed[F, R0])(f: R0 => R1)
+        : Managed[F, R1] =
+        new Mapped[F, E, R0, R1](fa, f)
+
+      // Note: the call to tailRecM should be tail recursive, however the returned `Managed` may still blow the stack
+      // when `use` is called.
```

Comment:
move it to the same line above

Reply:
I don't think it'll make a difference, as scaladocs on anonymous class instantiations are effectively invisible.